code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -28,8 +45,38 @@
 	qdel(src)
 	return crushed_can
 
+/obj/item/reagent_containers/food/drinks/cans/AltClick(mob/user)
+	var/mob/living/carbon/human/H
+	if(!can_shake || !ishuman(user))
+		return ..()
+	H = user
+	if(canopened)
+		to_chat(H,"<span class='notice'>You can't shake up an already opened drink!")
+		return ..()
+	if(src == H.l_hand || src == H.r_hand)
+		can_shake = FALSE
+		addtimer(CALLBACK(src, .proc/reset_shakable), 1 SECONDS)
+		to_chat(H, "<span class ='notice'>You start shaking up [src].</span>")
+		if(do_after(H, 1 SECONDS, target = H))
+			visible_message("<span class='warning'>[user.name] shakes up the [name]!</span>")
+			if(times_shaken == 0)
+				times_shaken++
+				shake_timer = addtimer(CALLBACK(src, .proc/reset_shaken), 1 MINUTES, TIMER_STOPPABLE)
```

Comment:
```suggestion
				addtimer(CALLBACK(src, .proc/reset_shaken), 1 MINUTES, TIMER_UNIQUE | TIMER_OVERRIDE | TIMER_NO_HASH_WAIT)
```
Using this will also make it behave the same way it does now using the existing timer system. I'd suggest using this to avoid extra code on your end.
The `TIMER_UNIQUE` says there should only be one timer of this sort. `TIMER_OVERRIDE` says to override the existing timer if any (AKA destroy it). `TIMER_NO_HASH_WAIT` will not include the `wait` parameter in the hashing method. Meaning any value of wait will result in the same hash thus is considered the same timer

Same goes for all the other `addtimer` calls